src/index.rs

```diff
@@ -36,6 +35,35 @@ pub struct IndexUpdate {
     pub added: HashMap<CanonicalPathBuf, ResourceId>,
 }
 
+impl IndexUpdate {
+    pub fn empty() -> Self {
+        IndexUpdate {
+            deleted: HashSet::new(),
+            added: HashMap::new(),
+        }
+    }
+
+    pub fn added(path: CanonicalPathBuf, id: ResourceId) -> Self {
+        let mut added = HashMap::new();
+        added.insert(path, id);
+
+        IndexUpdate {
+            deleted: HashSet::new(),
+            added,
+        }
+    }
+
+    pub fn deleted(id: ResourceId) -> Self {
+        let mut deleted = HashSet::new();
+        deleted.insert(id);
+
+        IndexUpdate {
+            deleted,
+            added: HashMap::new(),
+        }
+    }
```

Comment:
```suggestion
    pub fn added(mut self, path: CanonicalPathBuf, id: ResourceId) -> Self {
        self.added.insert(path, id);
        self
    }

    pub fn deleted(mut self, id: ResourceId) -> Self {
        self.deleted.insert(id);
        self
    }
```

and then we could use `IndexUpdate::default().added(path_buf, entry.id);`

Reply:
Oh yeah, that would be much better